flamingpy/utils/viz.py

```diff
@@ -271,28 +306,63 @@ def draw_EGraph(
         ax.axis("off")
     plt.tight_layout(pad=5)
     plt.draw()
-    return ax
+    return fig, ax
 
 
-def plot_binary_mat_heat_map(mat, show=True):
+@mpl.rc_context(plot_params)
+def plot_binary_mat_heat_map(mat, show=True, title=None):
+    """Plot the heat map of a binary matrix."""
+    fig = plt.figure()
+    ax = plt.gca()
+    if not isinstance(mat, np.ndarray):
+        mat = mat.toarray()
+    cmap = mpl.colors.ListedColormap(["C0", "C1"])
+    plt.matshow(mat, 0, cmap=cmap)
+    if title:
+        plt.title(title)
+    cbar = plt.colorbar(ticks=[0.25, 0.75])
+    cbar.ax.set_yticklabels(["0", "1"])
+    cbar.set_label(
+        "value", rotation=270, fontsize=plot_params.get("axes.labelsize", 10) * 1.2, labelpad=20
+    )
+    if show:
+        plt.show()
+
+    axs = [ax, cbar.ax]
+    return fig, axs
+
+
+@mpl.rc_context(plot_params)
+def plot_mat_heat_map(mat, show=True, title=None):
```

Comment:
What is the motivation for having two functions like this?

(Note in practice, the adjacency matrix of the EGraph will have a small number of discrete entries - 0, 1 or maybe 0, 1, -1. Which of those should be used? And the adjacency matrix of decoding objects will have many different entries corresponding to the edge weights.)

Reply:
Good point. This two functions can be merged and written for a more general use. 
I decided not to do so before because I didn't see any use case for that.